### DISCUSSION TITLE
How to transfer a parameter

### DISCUSSION THREAD

**Discussion by johans1234** [nguyenquyhy/Flight-Tracker-StreamDeck#168] [Q&A] [answered]
How do I transfer a parameter, e.g. for the toggle event ELEVATOR_SET.

How to set the elevator to the required level?   (-16383 - +16383)

Simulator= FS2020

**Comment by nguyenquyhy** [comment, marked as answer]
You can put the parameter into the Toggle parameter textbox. 
However, I am not sure what you are trying to achieve here. Setting elevator with a button feel strange.

**Reply by rmroc451** [reply]
maybe they want to do flaps 0/10/20/30 like a C172 with buttons down the side?

Neat idea!

**Reply by nguyenquyhy** [reply]
Hmm but this is ELEVATOR_SET for the elevator, not flap. Elevator is usually bound to an axis instead of being set a specific number, so I have a feeling they are looking for something else.

**Reply by rmroc451** [reply]
you are right, my eyes read flaps... good call.

**Reply by johans1234** [reply]
Thank you for your support. My question was more a generic one. There are some toggle events which allow or require to deliver a value (in the table with the toggle events shown in brackets). I am just wondering, how to deliver such a value as parameter for the toggle event.

back to my example/use case....

What I actually wanted to do is to set the elevator trim to a value of 0 (zero). I guess, the appropriate toggle event is AXIS_ELEV_TRIM_SET, but I am not able to set it to 0. But maybe I am just completely wrong with my understanding.

**Reply by nguyenquyhy** [reply]
`Toggle parameter` (below `Toggle event`) is where you can give value to the event. If you leave it empty, the value will be default at 0.

However, all axis events are a bit more tricky as there is usually some encoding involved. The value you pass to an event is always 0 to 4,294,967,295, so there must be some mappings from that range to the in-game value. 
I also heard that in FSX, the mapping also depends on sensitivity setting, but I have never actually tried those axis event myself since SimConnect has another way to set those value (i.e. setting the variable directly). Unfortunately the plugin does not support bypassing events and setting variables directly for generic buttons at the moment as it is a bit tricky to implement.
